lib/phoenix/live_dashboard/live/page_live.ex

```diff
@@ -106,12 +224,58 @@ defmodule Phoenix.LiveDashboard.PageLive do
   end
 
   @impl true
+  def handle_event("select_node", params, socket) do
+    param_node = params["node"]
+
+    node = Enum.find(nodes(), &(Atom.to_string(&1) == param_node))
+
+    page = socket.assigns.page
+
+    if node && node != page.node do
+      to = live_dashboard_path(socket, page.route, node, page.params)
+      {:noreply, push_redirect(socket, to: to)}
+    else
+      {:noreply, redirect_to_current_node(socket)}
+    end
+  end
+
+  def handle_event("select_refresh", params, socket) do
+    case Integer.parse(params["refresh"]) do
+      {refresh, ""} -> {:noreply, update_page(socket, refresh: refresh)}
+      _ -> {:noreply, socket}
+    end
+  end
```

Comment:
I think we can move this event and the one above to the MenuComponent as long as we make it stateful and pass phx-target. WDYT? To change the refresh we would need to do something like:

    send(self(), {:refresh, new_value})

WDYT?

Reply:
I think I did what you told me, but I'm not 100% sure. Let me know if it is not correct :)